src/mesh/mesh.h

```diff
@@ -357,9 +342,58 @@ class Mesh {
     /// Maximal distance of observe point from Mesh relative to its size
     double global_observe_radius() const;
 
-    /// Number of elements read from input.
-    unsigned int n_all_input_elements_;
+    /// Initialize element_vec_, set size and reset counters of boundary and bulk elements.
+    void init_element_vector(unsigned int size);
 
+    /// Returns range (of indexes in element_vector_) of boundary or bulk elements
+    inline pair<unsigned int, unsigned int> get_element_range(bool boundary=false) const {
+    	if (boundary) return make_pair<unsigned int, unsigned int>(element_vec_.size()-boundary_size_, element_vec_.size());
+    	else return make_pair<unsigned int, unsigned int>(0, bulk_size_+0);
+    }
+
+    /// Returns count of boundary or bulk elements
+    inline unsigned int n_elements(bool boundary=false) const {
+    	if (boundary) return boundary_size_;
+    	else return bulk_size_;
+    }
+
+    /// Begin iterator to bulk part of element vector
+    inline ElementIterator bulk_begin() {
```

Comment:
replace by ranges.

Reply:
Rename `Tools::GeneralIterator` to `Iter` and add there templated method make_iter for simplified cration of iterators from accessors. Try to implement also subtract() or overloaded - in accessors, so that the range class can return size().